packages/common/src/api/tan-query/events/useDeleteEvent.ts

```diff
@@ -36,8 +39,40 @@ export const useDeleteEvent = () => {
       // Update the event cache
       queryClient.setQueryData(getEventQueryKey(eventId), deletedEvent)
 
+      // Remove event from the list of events for the entity
+      let prevEntityState: ID[] = []
+      if (entityId) {
+        queryClient.setQueryData(
+          getEventsByEntityIdQueryKey({ entityId, entityType }),
+          (prevData) => {
+            const newState = cloneDeep(prevData) ?? []
+            prevEntityState = newState
+            newState.splice(newState.indexOf(eventId), 1)
+            return newState
+          }
+        )
+      }
+
+      // Remove event from the list of events for the entity by event type
+      let prevEventTypeState: ID[] = []
+      if (entityId) {
+        queryClient.setQueryData(
+          getEventsByEntityIdQueryKey({ entityId, entityType, eventType }),
+          (prevData) => {
+            const newState = cloneDeep(prevData) ?? []
```

Comment:
Do we need to clone?

Reply:
probably not anymore actually since it should be an array of numbers